rust/tools/authorization-logic/src/parsing/astconstructionvisitor.rs

```diff
@@ -38,8 +38,22 @@ pub fn parse_program(prog_text: &str) -> AstProgram {
     construct_program(&parse_result)
 }
 
+// The authorization logic frontend supports newlines
+// in the names of constants, but souffle does not. As
+// an easy way to support this, we delete newlines from
+// constant names. Since multiline literals/constants
+// are represented with triple quotes, we replace
+// triple quotes with single here.
+fn sanitize_id(name: String) -> String {
+    name.replace("\n", "").replace("\"\"\"", "\"")
+}
```

Comment:
> After reading more closely, I suspect the answer is "this will not cause issues, we want a multi-line constant with equivalent contents to a single-line constant to be treated identically."

Yes, I think the same way as you do.

I am not sure I follow your second comment. By the time the code in this file is running, antlr has already done all the parsing/lexing, and this piece of code is essentially the "earliest" piece that is not generated by antlr and is directly rust code that we are writing.

This sanitize_id function is essentially doing a very tiny and obvious part of the translation from auth logic into souffle early (just as we are building the AST objects rather than later when we do the rest of the translation). Perhaps your point was really not about doing this early, but about doing this later in case we want to handle the newlines differently later in the compilation pipeline... I don't see a use-case for this yet though, so I think doing this here is OK.

Let me know if I missed your point.

Reply:
I think your other comment in this thread about separating the quotes out in the antlr grammar is similar to the point you were making here. If so, perhaps this thread "is resolved" and the discussion can continue in the other related thread in `AuthLogic.g4`

Reply:
yes, that's correct.